controls/V-41600.rb

```diff
@@ -1,42 +1,69 @@
+  conditions change, the number and types of events to be logged may change, and
+  the web server must be able to facilitate these changes.
 
     The minimum list of logged events should be those pertaining to system
-startup and shutdown, system access, and system authentication events. If these
-events are not logged at a minimum, any type of forensic investigation would be
-missing pertinent information needed to replay what occurred.
+  startup and shutdown, system access, and system authentication events. If these
+  events are not logged at a minimum, any type of forensic investigation would be
+  missing pertinent information needed to replay what occurred.
+  "
+  desc  "check", "
+  Review the NGINX web server documentation and the deployed system configuration to determine 
+  if, at a minimum, system startup and shutdown, system access, and system authentication events are logged.
+    
+  Check for the following:
+  # grep the 'log_format' directive in the http context of the nginx.conf. 
+
+  The logs will not include the minimum logable events if the 'log_format' directive does not exist.
+
+  If the the 'log_format' directive does not exist, this is a finding.
+
+  Example configuration:
+  log_format  main  '$remote_addr - $remote_user [$time_local] ""$request""'
+  '$status $body_bytes_sent ""$http_referer""'
+  '""$http_user_agent"" ""$http_x_forwarded_for""';
+
   "
+  desc  "fix", "Configure the 'log_format' directive in the nginx.conf file to look like the following:
+
+  log_format  main  '$remote_addr - $remote_user [$time_local] ""$request""'
+  '$status $body_bytes_sent ""$http_referer""'
+  '""$http_user_agent"" ""$http_x_forwarded_for""';
+  
+  NOTE: Your log format may be using different variables based on your environment, 
+  however it should be verified to be producing the same end result of logged elements."
+
   impact 0.5
+  tag "severity": "medium"
   tag "gtitle": "SRG-APP-000089-WSR-000047"
   tag "gid": "V-41600"
   tag "rid": "SV-54177r3_rule"
   tag "stig_id": "SRG-APP-000089-WSR-000047"
   tag "fix_id": "F-47059r3_fix"
   tag "cci": ["CCI-000169"]
   tag "nist": ["AU-12 a", "Rev_4"]
-  tag "false_negatives": nil
-  tag "false_positives": nil
-  tag "documentable": false
-  tag "mitigations": nil
-  tag "severity_override_guidance": false
-  tag "potential_impacts": nil
-  tag "third_party_tools": nil
-  tag "mitigation_controls": nil
-  tag "responsibility": nil
-  tag "ia_controls": nil
-  tag "check": "Review the web server documentation and the deployed system
-configuration to determine if, at a minimum, system startup and shutdown,
-system access, and system authentication events are logged.
-
-If the logs do not include the minimum logable events, this is a finding."
-  tag "fix": "Configure the web server to generate log records for system
-startup and shutdown, system access, and system authentication events."
+
+  nginx_conf_handle = nginx_conf(input('conf_path'))
```

Comment:
call inputs directly in the control rather than making them a local instance variable